admin-ui/plugins/auth-server/components/Agama/AgamaAliasListPage.js

```diff
@@ -107,16 +135,40 @@ function AliasesListPage() {
     },
   })
 
-  const handleSubmit = useCallback(
-    (values) => {
+  const handleSubmit = useCallback(() => {
+    setShowCommitDialog(true)
+  }, [])
+
+  const handleDeleteAccept = (userMessage) => {
+    if (!rowToDelete) return
+    const userAction = {}
+    const postBody = {}
+    const value = { ...configuration?.acrMappings }
+    delete value[rowToDelete.mapping]
+    postBody['requestBody'] = [
+      {
+        path: '/acrMappings',
+        value: value,
+        op: configuration?.acrMappings ? 'replace' : 'add',
+      },
+    ]
+    buildPayload(userAction, userMessage, postBody)
+    dispatch(patchJsonConfig({ action: userAction }))
+    if (rowToDelete.resolve) rowToDelete.resolve(true)
+    setShowDeleteDialog(false)
```

Comment:
Implementing a manual Promise-like pattern in component state is bad because it violates the Keep It Simple, Stupid (KISS) principle by attaching resolver functions to a state object to manage control flow, which is an overly complex and unconventional pattern when a direct function call would achieve the same result more clearly.